src/main/java/ch/njol/skript/expressions/ExprTargetedBlock.java

```diff
@@ -36,55 +36,69 @@
 import org.eclipse.jdt.annotation.Nullable;
 
 @Name("Targeted Block")
-@Description("The block at the crosshair. This regards all blocks that are not air as fully solid, e.g. torches will be like a solid stone block for this expression.")
-@Examples({"# A command to set the block a player looks at to a specific type:",
-		"command /setblock &lt;material&gt;:",
-		"\ttrigger:",
-		"\t\tset targeted block to argument"})
+@Description({
+	"The block at the crosshair. This regards all blocks that are not air as fully solid, e.g. torches will be like a solid stone block for this expression.",
+	"The actual target block will regard the actual hit box of the block."
+})
+@Examples({
+	"set target block of player to stone",
+	"set target block of player to oak_stairs[waterlogged=true]",
+	"break target block of player using player's tool",
+	"give player 1 of type of target block",
+	"teleport player to location above target block",
+	"kill all entities in radius 3 around target block of player",
+	"set {_block} to actual target block of player",
+	"break actual target block of player"
+})
 @Since("1.0")
 public class ExprTargetedBlock extends PropertyExpression<Player, Block> {
 
 	static {
 		Skript.registerExpression(ExprTargetedBlock.class, Block.class, ExpressionType.COMBINED,
 				"[the] target[ed] block[s] [of %players%]", "%players%'[s] target[ed] block[s]",
-				"[the] actual[ly] target[ed] block[s] [of %players%]", "%players%'[s] actual[ly] target[ed] block[s]");
+				"[the] (actual[ly]|exact) target[ed] block[s] [of %players%]", "%players%'[s] (actual[ly]|exact) target[ed] block[s]");
 	}
 
+	private boolean actual;
+
 	@Override
 	@SuppressWarnings("unchecked")
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parser) {
 		setExpr((Expression<Player>) exprs[0]);
-		if (matchedPattern >= 2) {
-			// TODO remove 'actual' patterns in the future
-			Skript.warning("The 'actual' part of the targeted block expression is deprecated, it is now no longer required");
-		}
+		actual = matchedPattern > 1;
 		return true;
 	}
 
 	@Override
-	protected Block[] get(Event e, Player[] source) {
-		return get(source, p -> {
-			Block block = p.getTargetBlock(null, SkriptConfig.maxTargetBlockDistance.value());
-			if (block.getType() == Material.AIR)
+	protected Block[] get(Event event, Player[] source) {
+		Integer distance = SkriptConfig.maxTargetBlockDistance.value();
+		return get(source, player -> {
+			Block block;
+			if (actual)
+				block = player.getTargetBlockExact(distance);
+			else
+				block = player.getTargetBlock(null, distance);
+			if (block != null && block.getType() == Material.AIR)
```

Comment:
```suggestion
			if (block != null && block.getType() == Material.AIR)
```
Might be worth using https://github.com/SkriptLang/Skript/blob/11327579a9af912221ba6e0864d937935ead8799/src/main/java/ch/njol/skript/bukkitutil/ItemUtils.java#L106
for the air check